tiled/server/app.py

```diff
@@ -467,13 +468,20 @@ def override_get_settings():
                 settings.database_settings.max_overflow = database.max_overflow
             if database.init_if_not_exists is not None:
                 settings.database_init_if_not_exists = database.init_if_not_exists
-            if authenticators:
-                # If we support authentication providers, we need a database, so if one is
-                # not set, use a SQLite database in memory. Horizontally scaled deployments
-                # must specify a persistent database.
-                settings.database_settings.uri = (
-                    settings.database_settings.uri or "sqlite://"
-                )
+        if authenticators:
+            # If we support authentication providers, we need a database, so if one is
+            # not set, use a SQLite database in memory. Horizontally scaled deployments
+            # must specify a persistent database.
+            settings.database_settings.uri = (
+                settings.database_settings.uri or "sqlite://"
+            )
+            # If the only authentication provider is a ProxiedOIDCAuthenticator,
+            # set it on the settings object for downstream use.
+            if len(authenticators) == 1:
+                _, authenticator = next(iter(authenticators.items()))
```

Comment:
```suggestion
                authenticator = next(iter(authenticators.values()))
```

nit, it's a shame there's not an easier way to do this